codalab/model/bundle_model.py

```diff
@@ -519,6 +521,32 @@ def make_condition(key, field, value):
                     clause = cl_bundle.c.uuid.in_(
                         alias(select([cl_worksheet_item.c.bundle_uuid]).where(condition))
                     )
+            elif key == '.before':
+                target_datetime = parser.isoparse(value)
+                clause = cl_bundle.c.uuid.in_(
+                    alias(
+                        select([cl_bundle_metadata.c.bundle_uuid]).where(
+                            and_(
+                                cl_bundle_metadata.c.metadata_key == 'created',
+                                cl_bundle_metadata.c.metadata_value
+                                < int(target_datetime.timestamp()),
+                            )
+                        )
+                    )
+                )
+            elif key == '.after':
+                target_datetime = parser.isoparse(value)
+                clause = cl_bundle.c.uuid.in_(
+                    alias(
+                        select([cl_bundle_metadata.c.bundle_uuid]).where(
+                            and_(
+                                cl_bundle_metadata.c.metadata_key == 'created',
+                                cl_bundle_metadata.c.metadata_value
+                                >= int(target_datetime.timestamp()),
+                            )
+                        )
+                    )
+                )
```

Comment:
I wonder if it's possible to refactor these two conditions a bit so there's less duplication. Maybe factor out the inner comparison and change that based on the `key` somehow?

Reply:
Would it be much cleaner that way?

Reply:
It's mostly an aesthetic preference. That way it's a lot more obvious how the logic for the two cases differ, and there's less chance of inconsistency in the future if someone edits one case and not the other etc. But it probably doesn't matter that much.

Although another thing I just noticed is that before is `<` but after is `>=` That feels kind of inconsistent to me. Is there a reason one includes equality and one doesn't?

Reply:
Yeah, +1 to refactoring to reduce code duplication, especially if we're going to add `.date`.